azurelinuxagent/ga/exthandlers.py

```diff
@@ -842,14 +843,30 @@ def _download_extension_package(self, source_uri, target_file):
 
     def _unzip_extension_package(self, source_file, target_directory):
         self.logger.info("Unzipping extension package: {0}", source_file)
+        # Unzip the extension first in the temporary directory and then copy over to the target_directory
+        temp_dir = tempfile.mkdtemp(prefix=self.get_full_name())
         try:
-            zipfile.ZipFile(source_file).extractall(target_directory)
+            zipfile.ZipFile(source_file).extractall(temp_dir)
         except Exception as exception:
-            logger.info("Error while unzipping extension package: {0}", ustr(exception))
+            err_msg = "Error while unzipping extension package: {0}".format(ustr(exception))
+            self.logger.warn(err_msg)
+            self.report_event(message=err_msg, is_success=False, log_event=False)
             os.remove(source_file)
-            if os.path.exists(target_directory):
-                shutil.rmtree(target_directory)
+            if os.path.exists(temp_dir):
+                shutil.rmtree(temp_dir)
             return False
+
+        try:
+            copy_tree(temp_dir, target_directory)
```

Comment:
We never clean up the temp dir when extraction succeeds. From the [docs](https://docs.python.org/2/library/tempfile.html):

> The user of mkdtemp() is responsible for deleting the temporary directory and its contents when done with it.

Reply:
That's a good point! Thanks for catching it! I think I just thought that since its a temp directory the OS would eventually clean it up lol but I'll do it just to be sure.

Reply:
Done